hello-apl/default.nix

```diff
@@ -0,0 +1,17 @@
+{ stdenv, gnuapl }:
+
+stdenv.mkDerivation {
+  pname = "hello-apl";
+  version = "0.1.0";
+
```

Comment:
Curious on your thoughts on this one. [This article recommends the hack](https://nixos.wiki/wiki/Packaging/Quirks_and_Caveats) for single script files for a slight optimization, but I can see how it hinders readability.

Reply:
I think in a scenario where you truly only have one file you want to use, it's okay. But then you can just use `writeScriptBin` instead of `mkDerivation`.

Here, my goal is to demo how one might package any `apl` codebase, even if it has more than one file, so IMHO specifying `src` is preferrable.

Reply:
That makes a ton of sense, plus it'll be a good learning for me as well. Lemme migrate this to a Makefile-based package.